snuba/web/rpc/timeseries.py

```diff
@@ -0,0 +1,181 @@
+import uuid
+from datetime import datetime
+
+from google.protobuf.json_format import MessageToDict
+from sentry_protos.snuba.v1alpha.endpoint_aggregate_bucket_pb2 import (
+    AggregateBucketRequest,
+    AggregateBucketResponse,
+)
+
+from snuba.attribution.appid import AppID
+from snuba.attribution.attribution_info import AttributionInfo
+from snuba.datasets.entities.entity_key import EntityKey
+from snuba.datasets.entities.factory import get_entity
+from snuba.datasets.pluggable_dataset import PluggableDataset
+from snuba.query import SelectedExpression
+from snuba.query.conditions import combine_and_conditions, combine_or_conditions
+from snuba.query.data_source.simple import Entity
+from snuba.query.dsl import Functions as f
+from snuba.query.dsl import (
+    NestedColumn,
+    and_cond,
+    column,
+    in_cond,
+    literal,
+    literals_array,
+)
+from snuba.query.expressions import Expression, FunctionCall
+from snuba.query.logical import Query
+from snuba.query.query_settings import HTTPQuerySettings
+from snuba.request import Request as SnubaRequest
+from snuba.utils.metrics.timer import Timer
+from snuba.web.query import run_query
+
+_HARDCODED_MEASUREMENT_NAME = "eap.measurement"
+
+
+def _get_measurement_field(
+    request: AggregateBucketRequest,
+) -> Expression:
+    field = NestedColumn("attr_num")
+    # HACK
+    return field[request.metric_name]
+
+
+def _treeify_or_and_conditions(query: Query) -> None:
+    """
+    look for expressions like or(a, b, c) and turn them into or(a, or(b, c))
+                              and(a, b, c) and turn them into and(a, and(b, c))
+
+    even though clickhouse sql supports arbitrary amount of arguments there are other parts of the
+    codebase which assume `or` and `and` have two arguments
+
+    Adding this post-process step is easier than changing the rest of the query pipeline
+
+    Note: does not apply to the conditions of a from_clause subquery (the nested one)
+        this is bc transform_expressions is not implemented for composite queries
+    """
+
+    def transform(exp: Expression) -> Expression:
+        if not isinstance(exp, FunctionCall):
+            return exp
+
+        if exp.function_name == "and":
+            return combine_and_conditions(exp.parameters)
+        elif exp.function_name == "or":
+            return combine_or_conditions(exp.parameters)
+        else:
+            return exp
+
+    query.transform_expressions(transform)
+
+
+def _get_aggregate_func(
+    request: AggregateBucketRequest,
+) -> Expression:
+    FuncEnum = AggregateBucketRequest.Function
+    lookup = {
+        FuncEnum.FUNCTION_SUM: f.sum(
+            _get_measurement_field(request), alias="measurement"
```

Comment:
We set the alias of this as `aggregate_value`. I don't think it matters for simple queries, but it does matter for more complex formulas. I would change this for consistency.